lib/Test.pm6

```diff
@@ -489,7 +489,7 @@ multi sub like(
     $time_after = nqp::time_n;
     $got.defined; # Hack to deal with Failures
     my $ok;
-    if $got ~~ Str:D {
+    if Str($got) ~~ Str:D {
         my $test = $got ~~ $expected;
```

Comment:
Here, you're still smartmatching against the original, not its stringified variant. So `like <foo bar ber>, /^foo$/` would pass the test, despite there being no match, if `$got` is actually stringified.